agent/util-scripts/pbench-register-tool-trigger

```diff
@@ -69,13 +77,18 @@ while true; do
 	esac
 done
 
-if [ "$start_trigger" != "${start_trigger%%:*}" ]; then
+if [[ -z "$start_trigger" && -z "$stop_trigger" ]] ;then
+	printf "At least one of --start-trigger or --stop-trigger must be specified.\n"
```

Comment:
For now we might want to require both.  This will help catch problems, and any use that does not use one or the other can set it to a value that is not generated in the script output.

Reply:
What does it mean to have _no_ start trigger?  Wouldn't that mean that the tool never starts??

And, what does it mean to have no stop trigger...does the collection get stopped at the end of the iteration, or does it run longer?  (Can we end up with multiple copies of the same tool running concurrently in subsequent iterations??)

Reply:
@portante, should there be code in the Tool Meister to detect and report tools whose starts are never triggered?

Reply:
> What does it mean to have _no_ start trigger? Wouldn't that mean that the tool never starts??

It means that what ever entity ultimately call `pbench-tool-meister-start` would have to be calling `pbench-start-tools` directly, but stopping the tools when triggered.

> And, what does it mean to have no stop trigger...does the collection get stopped at the end of the iteration, or does it run longer? (Can we end up with multiple copies of the same tool running concurrently in subsequent iterations??)

The Tool Meister code won't start more tools until the previous set are stopped.

So if no stop trigger is provided, that just means the driving entity is smart enough to stop the tools themselves.

Reply:
Ah.  Tricky.

In that case, I don't see a good justification for requiring _both_...which leaves the awkward question of why we require _either one_.  (I suppose that the use case where the user would want _neither_ is so far out that it's worth his/her inconvenience in order to protect all the other users...which, I suppose, is why you're suggesting requiring both...eh?  But, the trick of supplying dummy text in the rare cases to work around the guards is kind of a hack...but, perhaps those cases are so rare that it's justified.)

Reply:
> which, I suppose, is why you're suggesting requiring both...eh?

Yes, thinking about this interface, which is already really difficult to get right, and is only really used by `pbench-specjbb` and `pbench-trafficgen` today, by requiring both we'll flag the really odd cases where they are not both specified, to help make sure the user really knows what they are doing.

That is to say, if the user is savvy enough to only provide start or stop triggers, and knows how to call start or stop tools directly, then they are savvy enough to provide dummy strings for the option that is ignored.  Savvy?  ;)

Reply:
Enlightening (or, to say it differently: savvy :-) ). Both are now required - more unit tests added.